### ISSUE TITLE
Data vs Type names

### ISSUE THREAD

**Issue by JonasDuregard** [bmillwood/haskell-src-meta#76]
It appears using `'()` specifically gives you the name of the data constructor `()`. Presumably `''()` gives the type name. In HSE, these names are identical. This gives errors for some very simple examples including this one: 
```
{-#LANGUAGE TemplateHaskell#-}
import Language.Haskell.Meta
$(either error return $ parseDecs "main :: IO ()\nmain = return ()")
```
Maybe a long term solution is to split ToName into ToTypeName and ToExpName (but this might not work if the code involves DataKinds), or maybe it can be solved using mkName and delegate the resolving of the name to TH.

**Comment by JonasDuregard** [comment]
Similarly but less common: `$(either error return $ parseDecs "nil :: [] a\nnil = []")` gives an error.

**Comment by JonasDuregard** [comment]
None of the tuple constructors are translated correctly it seems. Tested with:
`$(either error return $ parseDecs "pair :: (,) Int Int\npair = (,) 1 2")`

**Comment by JonasDuregard** [comment]
Fixed in a9a75a880c2bb15ad95686e4af96993e67b63695
